tests/unit/test_session.py

```diff
@@ -367,6 +367,15 @@ def test_credential_provider_not_called_when_creds_provided(self):
                          "explicit credentials were provided to the "
                          "create_client call.")
 
+class TestPerformOperation(BaseSessionTest):
+    def test_s3(self):
```

Comment:
Just that no exceptions are raised, i.e. is it possible to make a call with an operation. Apparently we didn't have any such test in place. Crazy!

I could write a bunch of tests for operation objects, but with their eminent removal it seems like a waste of time.